### ISSUE TITLE
No replacement for alternative models for Trident

### ISSUE THREAD

**Issue by Francy-Chan** [sp614x/optifine#3040]
Unlike bow and fishing rod, tridents doesn't have the replacements for trident_in_hand and trident_throwing.

**Comment by Nekzuris** [comment]
These can be edited like any other .json vanilla model, however when it's no longer in the players hand the CEM is missing.

**Comment by Francy-Chan** [comment]
Yes, I know, I'm talking about the replacement models, like bow charging and fishing rod cast. I need to have a different display json for when the player is charging the trident, otherwise it will looks backwards, but normal when just holding it. And I like the vanilla way to have a different flat icon in hud (using builtin/generated) than the model, so I can have a different model without breaking the vanilla aesthetics.

**Comment by Nekzuris** [comment]
I see, that doesn't require optifine, just use an override predicate in "trident_in_hand.json" like this
`"overrides": [
        {
            "predicate": {
                "throwing": 1
            },
            "model": "item/trident_throwing"
        }
    ]`
and change the display setting in "trident_throwing.json"

**Comment by Francy-Chan** [comment]
Then I can't dynamically change based on name and/or enchantment, or a very specific nbt tag.
